### PR TITLE
deblock: fix fate and conformance test

### PR DESCRIPTION
**Pull request by nuomi2021**
This PR will fix "make fate-vvc SAMPLES=$your_fate_dir" 
it also fixes the conformance test

### CODE REVIEW

File: libavcodec/vvc/filter.c [hunk 1]
```diff
@@ -835,10 +835,16 @@ static void vvc_deblock(const VVCLocalContext *lc, int x0, int y0, const int rs,
 
                 if (!all_zero_bs) {
                     uint8_t *src = vertical ? POS(c_idx, x, y) : POS(c_idx, y, x);
-                    if (!c_idx)
+                    if (!c_idx) {
                         fc->vvcdsp.lf.filter_luma[vertical](src, src_stride, beta, tc, no_p, no_q, max_len_p, max_len_q, horizontal_ctu_edge);
-                    else
-                        fc->vvcdsp.lf.filter_chroma[vertical](src, src_stride, beta, tc, no_p, no_q, max_len_p, max_len_q, vs);
+                    } else {
+                        int use_c = 1;      //always use c code only, since asm code is not ready
```
Comment (nuomi2021) [inline libavcodec/vvc/filter.c]: this code always use c code for deblock.
I tried 
```c
                        int use_c = vs;      //use c if we have shift
                        for (int i = 0;  i < DEBLOCK_STEP >> (2 - vs); i++) {
                            //use c if it's not weak
                            if (max_len_p[i] > 1 || max_len_q[i] > 1)
                                use_c = 1;
                        }
```

but the fate test is failed.
@stone-d-chen , could you help find some condition to make asm passed for fate?
thank you

Comment (stone-d-chen) [reply libavcodec/vvc/filter.c]: Hi @nuomi2021, this condition works
The condition is
- horizontal only (!vertical) 
- max_len_{p,q} != 0

```
if (!all_zero_bs) {
    uint8_t *src = vertical ? POS(c_idx, x, y) : POS(c_idx, y, x);
    if (!c_idx) {
        fc->vvcdsp.lf.filter_luma[vertical](src, src_stride, beta, tc, no_p, no_q, max_len_p, max_len_q, horizontal_ctu_edge);
    } else {
        if(!vertical && // horizontal only, and max_len_{p,q} cannot be 0
            (!vs && max_len_p[0] != 0 && max_len_q[0] != 0 && max_len_p[1] != 0 && max_len_q[1] != 0) &&
            (vs && max_len_p[0] != 0 && max_len_q[0] != 0 && max_len_p[1] != 0 && max_len_q[1] != 0 && 
                   max_len_p[2] != 0 && max_len_q[2] != 0 && max_len_p[3] != 0 && max_len_q[3] != 0 )) {
            fc->vvcdsp.lf.filter_chroma[vertical](src, src_stride, beta, tc, no_p, no_q, max_len_p, max_len_q, vs);
        } else {
            fc->vvcdsp.lf.filter_chroma_c[vertical](src, src_stride, beta, tc, no_p, no_q, max_len_p, max_len_q, vs);
        }
    }
}
```

Comment (nuomi2021) [reply libavcodec/vvc/filter.c]: vertical check is not needed since we did hook the vertical function.
fixed, please help review it again. 
thank you

**Review by stone-d-chen — COMMENTED** [review]
Looks good to me, conformance is passing for me.

**Pull request merged**